packages/communication-react/review/communication-react.api.md

```diff
@@ -1499,9 +1499,10 @@ export interface MessageThreadStrings {
 }
 
 // @public
-export interface MessageThreadStylesProps extends BaseCustomStylesProps {
+export interface MessageThreadStyles extends BaseCustomStylesProps {
     chatContainer?: ComponentSlotStyle;
-    chatMessageContainer?: ComponentSlotStyle;
+    chatItemMessageContainer?: (mine: boolean) => ComponentSlotStyle;
```

Comment:
Meta point: Each time we accept a function prop, we have to think about performance. 

Will this be called on each render, or just once?

* If it's called just once, it might be surprising to Contoso when changes in the return value (for whatever reason) aren't reflected. 
* If it is called on each render, Contoso needs to ensure that the function is cheap to call repeatedly (I guess in this case we don't need to worry about memoization because the styles will ultimated get transformed to already-memoized className via `mergeStyles`).

For style props, my recommendation would be to not accept function props. Instead, flatten out the options (of course this only works to a limited degree... if we start adding more axes of variability this won't scale)

```
localChatItemMessageCotainer?: ComponentSlotStyle
remoteChatItemMessageContainer?: ComponentSlotStyle
```

and so on.

The reason I think functions are not a good idea here is that styles aren't intended to be updated dynamically, so our API should be optimized for that.

Reply:
ok thats a good idea. I'll separate it out and then we can swap between the two different styles in the component based on the mine property but at least we're not evaluate new style objects each time.